xonsh/ansi_colors.py

```diff
@@ -43,36 +43,39 @@ def partial_color_format(template, style='default', cmap=None, hide=False):
     colon = ':'
     expl = '!'
     toks = []
-    for literal, field, spec, conv in formatter.parse(template):
-        toks.append(literal)
-        if field is None:
-            pass
-        elif field in cmap:
-            toks.extend([esc, cmap[field], m])
-        elif '#' in field:
-            field = field.lower()
-            pre, _, post = field.partition('#')
-            f_or_b = '38' if RE_BACKGROUND.search(pre) is None else '48'
-            rgb, _, post = post.partition('_')
-            c256, _ = rgb_to_256(rgb)
-            color = f_or_b + ';5;' + c256
-            mods = pre + '_' + post
-            if 'underline' in mods:
-                color = '4;' + color
-            if 'bold' in mods:
-                color = '1;' + color
-            toks.extend([esc, color, m])
-        elif field is not None:
-            toks.append(bopen)
-            toks.append(field)
-            if conv is not None and len(conv) > 0:
-                toks.append(expl)
-                toks.append(conv)
-            if spec is not None and len(spec) > 0:
-                toks.append(colon)
-                toks.append(spec)
-            toks.append(bclose)
-    return ''.join(toks)
+    try:
```

Comment:
Yeah, I understand that this is a little ugly.  I can refactor a bit, since I think the only piece we probably really need to wrap is the initial call to the parser.  But I do think it is worth avoiding having to wrap every call in try/except.